spec/integration/winrm_spec.rb

```diff
@@ -35,14 +33,24 @@
       expect(result['_error']['msg']).to eq('The command failed with exit code 1')
     end
 
-    it 'runs a task', :reset_puppet_settings do
-      result = run_one_node(%W[task run #{stdin_task} message=somemessage] + config_flags)
-      expect(result['_output'].strip).to match(/STDIN: {"messa/)
+    it 'runs a task reading from stdin', :reset_puppet_settings do
+      result = run_one_node(%w[task run sample::winstdin message=❄somemessage] + config_flags)
```

Comment:
@Iristyle these tests wouldn't have worked before either. They work locally with Nano Server, since it defaults to a UTF-8 encoding/codepage. I'm having trouble finding the right way to reset it in AppVeyor so I can have a consistent target codepage for testing.

Reply:
Something funny's happening with this output. Not sure how to fix it.

Reply:
I can't get these tests to work with Windows Server 2012 R2 with either the old or new code. The unicode doesn't round-trip through the script execution correctly.